hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -387,22 +387,27 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	reportLimitedSuportEnabled(hcluster)
-	reportSilecedAlerts(hcluster)
+	reportSilencedAlerts(hcluster)
 	reportProxyConfig(hcluster)
 
 	// If deleted, clean up and return early.
 	if !hcluster.DeletionTimestamp.IsZero() {
 		reportHostedClusterGuestCloudResourcesDeletionDuration(hcluster)
+		reportHostedClusterDeletionDuration(hcluster, r.Clock)
```

Comment:
why do we need to report deletion before the hostedCluster has actually been deleted?

Reply:
This metric mesure the time it took to delete the cluster (once the cluster is deleted) or the elapsed time since the cluster is deleting.
The best is to set it as soon as we know the cluster is deleting.
Of course this method need to be called a second time, after removing the finalizer, to make sure we properly account the time it took to delete the cluster.

Reply:
makes sense. just to confirm, `metricVec.DeletePartialMatch(labelsToMatch)` will stop Prometheus from scraping this metric and creating more timeseries, and the old data points will still be there, right?

Reply:
cc @enxebre any objections/thoughts?

Reply:
`metricVec.DeletePartialMatch(labelsToMatch)` tells to end the time series matching the given labels.
The old points are not removed (they have already been scrapped by Prometheus).
As a general remark, please note that actions on the `MetricVec` object have no direct incidence on the data which are scrapped as Prometheus scrapes the `hypershift-operator` at its own pace.

Reply:
Sounds reasonable, this differs from what we do for any other metric like hypershift_cluster_available_duration_seconds or hypershift_cluster_initial_rollout_duration_seconds which will be confusing, at minimum we need to update the metric description for hypershift_cluster_deletion_duration_seconds if we move forward with this. Alternative we could capture when a deletionTimestamp is set to get this information.

cc @csrwng

Reply:
Yes, it is true that this is not very consistent with those other metrics.
An other metric that come into my mind is `hypershift_cluster_upgrade_duration_seconds` which is only set when the upgrade is finished.
I actually think that all those metrics should only bet set while doing the work; i.e.:
- While deleting a cluster for `hypershift_cluster_deletion_duration_seconds`
- While deleting resources used by the cluster for `hypershift_cluster_guest_cloud_resources_deletion_duration_seconds`
- While waiting the HCP to be available for `hypershift_cluster_available_duration_seconds`
- While waiting the whole hosted cluster to be "completed" for `hypershift_cluster_initial_rollout_duration_seconds`
- While upgrading for `hypershift_cluster_upgrade_duration_seconds`.

Setting those metrics once the work is finished is not actually helpful; and using the proper `last_over_time{metric[28d]}` Prometheus expression, there is always a way to continue / extend the metric over time if needed.

I was actually planning to submit a PR after this one to change the semantic of those metrics and make sure we measure the time only while doing the work and not after it is done.  
This PR will also move all the `report` methods in `hosted_controller.go` to the corresponding `metrics.go` file. 

Last but not least, remark that metric `hypershift_cluster_guest_cloud_resources_deletion_duration_seconds` is already measured while doing the work; but it is also measured after the work is done which might be confusing

And yes, if needed I can change the description of `hypershift_cluster_deletion_duration_seconds` to highlight the fact that it is now measured while deleting and not once deleted. I can also change the description of `hypershift_cluster_guest_cloud_resources_deletion_duration_seconds` to reflect that it is both measured at deleting time and once deleted.

Reply:
@enxebre : hello, PR has been updated to reflect that semantic change.